File: AnimationProject/app/src/main/java/com/example/animationproject/MainActivity.java
package com.example.animationproject;

import android.animation.Animator;
import android.animation.TimeAnimator;
import android.animation.TimeInterpolator;
import android.app.Activity;
import android.graphics.drawable.AnimationDrawable;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.view.ViewPropertyAnimator;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.IOException;
import java.util.Random;
import java.util.Timer;

public class MainActivity extends AppCompatActivity {

    MediaPlayer mp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mp = MediaPlayer.create(this, R.raw.music);
        mp.start();

        ImageView dancers1 = findViewById(R.id.dancers1);
        ImageView dancers2 = findViewById(R.id.dancers2);
        ImageView dancers3 = findViewById(R.id.dancers3);

        ViewPropertyAnimator vpa = dancers1.animate();
        vpa.setStartDelay(5500).alpha(1F);

        //dancers2.animate().setStartDelay(34000).alpha(1F);
        dancers3.animate().setStartDelay(40000).alpha(1F);

        ((AnimationDrawable) dancers1.getDrawable()).start();
        ((AnimationDrawable) dancers2.getDrawable()).start();
        ((AnimationDrawable) dancers3.getDrawable()).start();

        Thread thread = new Thread(){
            @Override
            public void run() {
                try {
                    Thread.sleep(34000);
                }catch (InterruptedException e){
                }

                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        dancers2.setVisibility(View.VISIBLE);
                    }
                });
            }

        };
        thread.start();
    }

}

File: CriminalIntent/app/src/main/java/com/bignerdranch/android/criminalintent/MainActivity.kt
package com.bignerdranch.android.criminalintent

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import java.util.UUID

private const val TAG = "MainActivity"
class MainActivity : AppCompatActivity(), CrimeListFragment.Callbacks{

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val currentFragmentID = R.id.fragment_container
        //val currentFragment = supportFragmentManager.findFragmentById(R.id.fragment_container)
        val currentFragment = supportFragmentManager.findFragmentById(currentFragmentID)

        if (currentFragment == null){
            val fragment = CrimeListFragment.newInstance()
            supportFragmentManager.beginTransaction()
                //.add(R.id.fragment_container, fragment)
                .add(currentFragmentID, fragment)
                .commit()
        }
    }

    override fun onCrimeSelected(crimeId: UUID) {
        val fragment = CrimeFragment.newInstance(crimeId)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .addToBackStack(null)
            .commit()
    }
}